letter-sender/src/test/java/com/christmas/letter/sender/service/SnsServiceTest.java

```diff
@@ -0,0 +1,94 @@
+package com.christmas.letter.sender.service;
+
+import com.christmas.letter.sender.exceptions.InvalidDataException;
+import com.christmas.letter.sender.model.Letter;
+import com.christmas.letter.sender.model.Location;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.boot.test.context.SpringBootTest;
+import software.amazon.awssdk.services.sns.SnsClient;
+import software.amazon.awssdk.services.sns.model.PublishRequest;
+import software.amazon.awssdk.services.sns.model.PublishResponse;
+
+import static org.hamcrest.MatcherAssert.assertThat;;
+
+import static org.hamcrest.Matchers.notNullValue;
+import static org.junit.Assert.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.*;
+
+@SpringBootTest
```

Comment:
@SpringBootTest is used for integration testing, however, this is clearly a unit test - you are mocking the dependencies and testing the service in isolation. So, you can you @ExtendWith(MockitoExtension.class

Reply:
Removed @SpringBootTest use @ExtendWith(MockitoExtension.class)